modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java

```diff
@@ -276,6 +280,54 @@ private String _buildFragmentHTML(
 		return sb.toString();
 	}
 
+	private JSONArray _getActionsJSONArray(
+			ObjectDefinition fdsViewObjectDefinition,
+			ObjectEntry fdsViewObjectEntry)
+		throws Exception {
+
+		Set<ObjectEntry> fdsActionObjectEntries = new TreeSet<>(
+			new ObjectEntryComparator(
+				ListUtil.toList(
+					ListUtil.fromString(
+						MapUtil.getString(
+							fdsViewObjectEntry.getProperties(),
+							"fdsActionsOrder"),
+						StringPool.COMMA),
+					Long::parseLong)));
+
+		fdsActionObjectEntries.addAll(
+			_getRelatedObjectEntries(
+				fdsViewObjectDefinition, fdsViewObjectEntry,
+				"fdsViewFDSActionRelationship"));
+
+		return JSONUtil.toJSONArray(
+			fdsActionObjectEntries,
+			(ObjectEntry fdsActionObjectEntry) -> {
+				Map<String, Object> properties =
+					fdsActionObjectEntry.getProperties();
+
+				return JSONUtil.put(
+					"data",
+					JSONUtil.put(
+						"confirmationMessage",
+						properties.get("confirmationMessage")
```

Comment:
This should work no matter what we input in the confirmation message text box, I mean, regardless localization is off or on for object fields, our reference is `confirmationMessage`

Reply:
This is the OOTB API! I was surprised it works, since the frontend part so is awkward, with the [separate _i18n fields](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/actions/ItemActionForm.tsx#L116-L119).


https://github.com/liferay-frontend/liferay-portal/assets/5435511/0c64da3c-7af1-425c-a7ba-d3eef0227a7d